xla/service/gpu/transforms/collective_backend_assigner.h

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2025 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_TRANSFORMS_COLLECTIVE_BACKEND_ASSIGNER_H_
+#define XLA_SERVICE_GPU_TRANSFORMS_COLLECTIVE_BACKEND_ASSIGNER_H_
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/statusor.h"
+#include "absl/log/log.h"
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/hlo/pass/hlo_pass_interface.h"
+
+namespace xla {
+namespace gpu {
+
+class CollectiveBackendAssigner : public HloModulePass {
+ public:
+  explicit CollectiveBackendAssigner(int64_t threshold_in_bytes)
+      : threshold_in_bytes_(threshold_in_bytes) {}
+
+  absl::string_view name() const override {
+    return "collective-backend-assigner";
+  }
+
+  absl::StatusOr<bool> Run(
+      HloModule* module,
+      const absl::flat_hash_set<absl::string_view>& execution_threads) override;
+
+  static bool HasInternodeCommunication(
```

Comment:
Nit: for clarity it would be better to have only function headers in this file, and the implementation in the cc file.